src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.Windows.cs

```diff
@@ -82,5 +80,28 @@ internal static unsafe int StringToAnsiString(string s, byte* buffer, int buffer
             buffer[nb] = 0;
             return nb;
         }
+
+        // Returns number of bytes required to convert given string to Ansi string. The return value includes null terminator.
+        internal static unsafe int GetAnsiStringByteCount(ReadOnlySpan<char> chars)
```

Comment:
There is more consolidation that can be done using these two methods in the interop code, but I had to stop somewhere...